client/src/partials/fiscal/templates/modals/fiscal.closing.modal.js

```diff
@@ -0,0 +1,117 @@
+angular.module('bhima.controllers')
+  .controller('ClosingFiscalYearModalController', ClosingFiscalYearModalController);
+
+// dependencies injection
+ClosingFiscalYearModalController.$inject = [
+  'AccountService', 'NotifyService', 'FiscalService', 'ModalService', '$uibModalInstance', 'data'
+];
+
+// The closing fiscal year controller
+function ClosingFiscalYearModalController(Accounts, Notify, Fiscal, Modal, Instance, Data) {
+  var vm = this;
+
+  // global variables
+  vm.steps  = 'default';
+
+  // expose to the view
+  vm.cancel = Instance.close;
+  vm.stepForward = stepForward;
+
+  // load the list of accounts
+  Accounts.read()
+  .then(function (accounts) {
+    vm.accounts = accounts;
+  })
+  .catch(Notify.handleError);
+
+  Fiscal.read(Data.id)
+  .then(function (fiscal) {
+    vm.fiscal = fiscal;
+
+    // get balance of period N of the year to close
+    return Fiscal.periodicBalance({
+      id: vm.fiscal.id,
+      period_number: vm.fiscal.number_of_months // last month
+    });
+  })
+  .then(function (balance) {
+
+    vm.exploitation = { profit: [], charge: [] };
+
+    balance.forEach(function (item) {
+
+      if (item.type === 'income') {
+        vm.exploitation.profit.push(item);
+      }
+
+      if (item.type === 'expense') {
+        vm.exploitation.charge.push(item);
+      }
+
+    });
+
+    return vm.exploitation;
+
+  })
+  .then(exploitationStatus)
```

Comment:
Why does this need to be in a `.then()` method?

Can you not just do 
```js
// calculate vm.exploitation

 vm.profit = creditorSold(vm.exploitation.profit);
 vm.charge = debitorSold(vm.ploitation.charge);
 vm.globalResult = (vm.profit - vm.charge);
}).catch( /* etc .. */);
```
without writing an extra `.then()` call?